py/nlr.h

```diff
@@ -89,24 +103,26 @@
     #define MICROPY_NLR_MIPS (1)
     #define MICROPY_NLR_NUM_REGS (MICROPY_NLR_NUM_REGS_MIPS)
 #else
-    #define MICROPY_NLR_SETJMP (1)
-    //#warning "No native NLR support for this arch, using setjmp implementation"
-#endif
+    #if defined(MICROPY_NLR_SETJMP)
+        #error "No native NLR support for this arch, and setjmp was disabled."
+    #else
+        // Enable the fall-back setjump implementation.
+        #define MICROPY_NLR_SETJMP (1)
+        // #warning "No native NLR support for this arch, using setjmp implementation"
```

Comment:
Should we just delete this warning if we are not going to enable it?

Reply:
I think so too... I'll leave it to @dpgeorge to decide.